src/k8-types/src/app/deployment.rs

```diff
@@ -0,0 +1,84 @@
+use serde::Deserialize;
+use serde::Serialize;
+
+use crate::core::pod::PodSpec;
+use crate::{Crd, CrdNames, DefaultHeader, LabelSelector, Spec, Status, TemplateSpec};
+
+const DEPLOYMENT_API: Crd = Crd {
+    group: "apps",
+    version: "v1",
+    names: CrdNames {
+        kind: "Deployment",
+        plural: "deployments",
+        singular: "deployment",
+    },
+};
+
+#[derive(Deserialize, Serialize, Debug, Default, Clone, PartialEq)]
+#[serde(rename_all = "camelCase", default)]
+pub struct DeploymentSpec {
+    pub min_ready_seconds: Option<i32>,
+    pub paused: Option<bool>,
+    pub progress_deadline_seconds: Option<i32>,
+    pub replicas: Option<i32>,
+    pub revision_history_limit: Option<i32>,
+    pub selector: LabelSelector,
+    pub strategy: Option<DeploymentStrategy>,
+    pub template: TemplateSpec<PodSpec>,
+}
+
+#[derive(Deserialize, Serialize, Debug, Default, Clone, PartialEq)]
+#[serde(rename_all = "camelCase", default)]
+pub struct DeploymentStrategy {
+    pub rolling_update: Option<RollingUpdateDeployment>,
+    #[serde(rename = "type")]
+    pub type_: Option<String>,
+}
+
+#[derive(Deserialize, Serialize, Debug, Default, Clone, PartialEq)]
+#[serde(rename_all = "camelCase", default)]
+pub struct RollingUpdateDeployment {
+    pub max_surge: Option<String>,
+    pub max_unavailable: Option<String>,
+}
+
+impl Spec for DeploymentSpec {
+    type Status = DeploymentStatus;
+    type Header = DefaultHeader;
+
+    fn metadata() -> &'static Crd {
+        &DEPLOYMENT_API
+    }
+
+    fn make_same(&mut self, _other: &Self) {
+        todo!();
```

Comment:
@sehz I'm not sure what to do here.

Reply:
This is optional where you need custom logic to deal with k8 strategic merge.  Just remove it